app/data_providers/dispatch.py

```diff
@@ -82,27 +84,56 @@ def __init__(
             corpus_client: CorpusFeatureGroupClient,
             user_recommendation_preferences_provider: UserRecommendationPreferencesProvider,
             topic_provider: TopicProvider,
+            personalized_for_you_slate_provider: PersonalizedForYouSlateProvider,
             topic_slate_provider: TopicSlateProvider,
             collection_slate_provider: CollectionSlateProvider,
     ):
         self.topic_provider = topic_provider
         self.corpus_client = corpus_client
         self.user_recommendation_preferences_provider = user_recommendation_preferences_provider
+        self.personalized_for_you_slate_provider = personalized_for_you_slate_provider
         self.topic_slate_provider = topic_slate_provider
         self.collection_slate_provider = collection_slate_provider
 
     async def get_slate_lineup(
             self, user: UserIds, slate_count: int, recommendation_count: int
     ) -> CorpusSlateLineupModel:
+        """
+        Returns the Home slate lineup:
+        1. 'For You' slate if preferred topics are available, otherwise this slate is simply not shown.
+        2. Collection slate
+        3. Topic slates according to preferred topics if available, otherwise default topics.
+
+        :param user:
+        :param slate_count:
+        :param recommendation_count:
+        :return:
+        """
+        slates = []
+
+        preferred_topics = await self._get_preferred_topics(user)
+        if preferred_topics:
+            slates += [self.personalized_for_you_slate_provider.get_slate(preferred_topics, recommendation_count)]
+
+        slates += [
+            self.collection_slate_provider.get_slate(),
+            self._get_topic_slates(preferred_topics=preferred_topics, recommendation_count=recommendation_count),
+        ]
+
         return CorpusSlateLineupModel(
             slates=flatten(list(
-                await gather(
-                    self.collection_slate_provider.get_slate(),
-                    self._get_topic_slates(recommendation_count=recommendation_count),
-                )
+                await gather(*slates)
             )),
         )
 
-    async def _get_topic_slates(self, recommendation_count: int) -> List[CorpusSlateModel]:
-        topics = await self.topic_provider.get_topics(self.DEFAULT_TOPICS)
+    async def _get_preferred_topics(self, user: UserIds) -> List[TopicModel]:
+        user_recommendation_preferences = await self.user_recommendation_preferences_provider.fetch(str(user.user_id))
```

Comment:
nit: long names are not pythonic style, harder to read and pollute the code. smth like 'user_prefs' would work here